catalog/transformer/catalog-transformer-geojson-queryresponse/src/main/java/ddf/catalog/transformer/queryresponse/geojson/GeoJsonQueryResponseTransformer.java

```diff
@@ -63,17 +66,33 @@
     }
   }
 
-  public static JSONObject convertToJSON(Result result) throws CatalogTransformerException {
+  private MetacardTransformer metacardTransformer;
+
+  private JSONObject convertToJSON(Result result) throws CatalogTransformerException {
     JSONObject rootObject = new JSONObject();
 
     addNonNullObject(rootObject, "distance", result.getDistanceInMeters());
     addNonNullObject(rootObject, "relevance", result.getRelevanceScore());
-    addNonNullObject(
-        rootObject, "metacard", GeoJsonMetacardTransformer.convertToJSON(result.getMetacard()));
+    addNonNullObject(rootObject, "metacard", processMetacard(result.getMetacard()));
 
     return rootObject;
   }
 
+  private JSONObject processMetacard(Metacard metacard) throws CatalogTransformerException {
+    if (metacardTransformer == null) {
+      throw new CatalogTransformerException("No metacard transformer with id=geojson is available");
+    }
+
+    BinaryContent rawContent = metacardTransformer.transform(metacard, null);
+    JSONParser jsonParser = new JSONParser(JSONParser.DEFAULT_PERMISSIVE_MODE);
+    try {
+      return (JSONObject) jsonParser.parse(rawContent.getInputStream());
```

Comment:
Should we check the `instanceof` before casting?

Reply:
After looking more into the docs, `jsonParser.parse()` can either return a JSONObject, JSONArray, String, or a primitive type, all of which could be a valid value for the `metacard` field (I don't anticipate anyone wanting to set `metacard` to 1, true, or some string but that's beside the point). To make this as generic as possible, I don't think any type casting should be used at all. `addNonNullObject(rootObject, "metacard", processMetacard(result.getMetacard()))` can handle all of these cases.